gax-httpjson/src/test/java/com/google/api/gax/httpjson/testing/MockHttpService.java

```diff
@@ -0,0 +1,163 @@
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.httpjson.testing;
+
+import com.google.api.client.http.LowLevelHttpRequest;
+import com.google.api.client.http.LowLevelHttpResponse;
+import com.google.api.client.testing.http.MockHttpTransport;
+import com.google.api.client.testing.http.MockLowLevelHttpRequest;
+import com.google.api.client.testing.http.MockLowLevelHttpResponse;
+import com.google.api.gax.httpjson.ApiMessage;
+import com.google.api.gax.httpjson.ApiMethodDescriptor;
+import com.google.api.pathtemplate.PathTemplate;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+
+/* Mocks an HTTPTransport. Expected responses and exceptions can be added to a queue
+ * from which this mock HttpTransport polls when it relays a response. */
+public final class MockHttpService extends MockHttpTransport {
+
+  private final List<String> requestPaths = new LinkedList<>();
+  private final Queue<HttpResponseFactory> responseHandlers = new LinkedList<>();
+  private List<ApiMethodDescriptor<? extends ApiMessage, ? extends ApiMessage>> serializers;
+  private String endpoint;
+
+  /* Create a MockHttpService.
+   *
+   * @param serializers - the list of method descriptors for the methods that the mocked server supports.
+   * @param fixedEndpoint - the fixed portion of the endpoint URL that prefixes the methods' path template substring. */
+  public MockHttpService(
+      List<ApiMethodDescriptor<? extends ApiMessage, ? extends ApiMessage>> serializers,
+      String fixedEndpoint) {
```

Comment:
I think `pathPrefix` would be more correct than `fixedEndpoint`

Reply:
Done.